pilot/pkg/bootstrap/mesh.go

```diff
@@ -51,6 +61,19 @@ func (s *Server) initMeshConfiguration(args *PilotArgs, fileWatcher filewatcher.
 	s.environment.Watcher = mesh.NewFixedWatcher(&meshConfig)
 }
 
+// maybeWatchMeshConfiguration watches mesh configuration from a ConfigMap if no local file is present.
+// This is done after initMeshConfiguration, which must be done to initialize the Kubernetes client.
+func (s *Server) maybeWatchMeshConfiguration(args *PilotArgs) {
+	if s.kubeClient == nil {
+		return
+	}
+	if _, err := os.Stat(args.MeshConfigFile); os.IsNotExist(err) {
+		configMapName := getConfigMapName(args.Revision)
+		s.environment.Watcher = mesh.NewConfigMapWatcher(
+			s.kubeClient, constants.IstioSystemNamespace, configMapName, configMapKey)
```

Comment:
we should use the passed in namespace in `args`. `IstioSystemNamespace` is only the default